metaflow/plugins/airflow/sensors/external_task_sensor.py

```diff
@@ -0,0 +1,96 @@
+from .base_sensor import AirflowSensorDecorator
+from ..airflow_utils import SensorNames
+from ..exception import AirflowException
+from datetime import timedelta
+
+
+AIRFLOW_STATES = dict(
+    QUEUED="queued",
+    RUNNING="running",
+    SUCCESS="success",
+    SHUTDOWN="shutdown",  # External request to shut down,
+    FAILED="failed",
+    UP_FOR_RETRY="up_for_retry",
+    UP_FOR_RESCHEDULE="up_for_reschedule",
+    UPSTREAM_FAILED="upstream_failed",
+    SKIPPED="skipped",
+)
+
+
+class ExternalTaskSensorDecorator(AirflowSensorDecorator):
+    operator_type = SensorNames.EXTERNAL_TASK_SENSOR
```

Comment:
Because `SensorNames` is used in `airflow_utils.py`. Some parts can't move from airflow_utils if they are used in instantiating airflow-related objects. We create an `AirflowTask`(internal object) from the decorator but we build the airflow-related objects from `airflow_utils` file. This file uses SensorNames created via the decorator.